hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java

```diff
@@ -291,10 +293,11 @@ private PreHandleResult expandAndVerifySignatures(
         // 2b. Call Pre-Transaction Handlers
         try {
             // First, perform semantic checks on the transaction
-            dispatcher.dispatchPureChecks(txInfo.txBody());
+            dispatcher.dispatchPureChecks(txBody);
             // Then gather the signatures from the transaction handler
             dispatcher.dispatchPreHandle(context);
-            // FUTURE: Finally, let the transaction handler do warm up of other state it may want to use later (TBD)
+            // Finally, let the transaction handler do warm up of other state it may want to use later
+            dispatcher.dispatchWarmup(context);
```

Comment:
So, we need `warmCache` on each store now ? So we can call underlying `VirtualMap.warm`?

Reply:
Yes, at least for the entities we want to warm up.